server/service/src/static_files.rs

```diff
@@ -135,6 +149,130 @@ impl StaticFileService {
             path: file_path.to_string_lossy().to_string(),
         }))
     }
+
+    pub async fn download_file_from_central(
+        &self,
+        id: &str,
+        category: StaticFileCategory,
+        service_provider: &ServiceProvider,
+    ) -> anyhow::Result<Option<StaticFile>> {
+        log::info!("Downloading file from central server");
+        if is_central_server() {
+            return Err(anyhow::Error::msg(
+                "Can't download file from central server, as I am the central server!",
+            ));
+        }
+
+        let (table_name, record_id) = match category.clone() {
+            StaticFileCategory::SyncFile(table_name, record_id) => (table_name, record_id),
+            _ => {
+                return Err(anyhow::Error::msg(
+                    "Can't download file from central server, as it's not a sync file!",
+                ))
+            }
+        };
+
+        let ctx = service_provider
+            .context("NoStore".to_string(), SYSTEM_USER_ID.to_string())
+            .map_err(|_| anyhow::Error::msg("Can't create context"))?;
+        let settings = service_provider
+            .settings
+            .sync_settings(&ctx)
+            .map_err(|_| anyhow::Error::msg("Can't access sync settings"))?;
+
+        let base_url = match settings {
+            Some(settings) => settings.file_upload_base_url(),
+            None => {
+                return Err(anyhow::Error::msg(
+                    "Can't download file as sync is not configured",
+                ))
+            }
+        };
+
+        let sync_file_repo = SyncFileReferenceRowRepository::new(&ctx.connection);
+
+        let sync_file_ref = sync_file_repo.find_one_by_id(&id)?;
+
+        let sync_file_ref = match sync_file_ref {
+            Some(sync_file_ref) => sync_file_ref,
+            None => {
+                return Err(anyhow::Error::msg(format!(
+                    "Can't find sync file reference with id: {}",
+                    id
+                )))
+            }
+        };
+
+        let file_name = sync_file_ref.file_name.clone();
+        let download_url = format!("{}/{}/{}?id={}", base_url, table_name, record_id, id);
```

Comment:
I feel like download and upload from central is very similar to sync api v6 pattern, I wonder if we can make it stronger type by using the same pattern ?

Sorry it would mean changing the logic a bit, I'll try to have a go at showing what mean in a suggestion.